code/go/0chain.net/mocks/core/datastore/Store.go

```diff
@@ -0,0 +1,197 @@
+// Code generated by mockery 2.7.4. DO NOT EDIT.
+
+package mocks
```

Comment:
Should this not be `package datastore`?

Reply:
I can not replace the `mocks` package with the `datastore` package, because it is generated files and `mockery` doesn't provide ability to changing package in generated files. Sure, we can do it manually, but it is a bad practice - editing generated files.
I can replace the `datastore` package with the `mocks` package in PR 236. If there is a circular dependency in future tests, we can easily isolate these tests into a separate `*_test` package and use black-box testing, I think.

> package naming should be consistent between this Pr and sharder unit tests #236 and also [batchI.go](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/mocks/core/persistencestore/BatchI.go#L3), [IteratorI.go](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/mocks/core/persistencestore/IteratorI.go#L3), [QueryI.go](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/mocks/core/persistencestore/QueryI.go#L3) and [SessionI.go](https://github.com/0chain/0chain/blob/master/code/go/0chain.net/mocks/core/persistencestore/SessionI.go#L3) which on reflection should probably all be in `persistencestore`.
> 
> I see there is a potential problem with circular dependencies. Ideas
> 
> 1. https://github.com/golang/go/wiki/CodeReviewComments#import-dot .  So the tests are put in a different package so they can import the mocks, dot here is being used to pretend the tests are in the package they are testing.
> 2. An alternative would be to more aggressive about splitting up interface and implementation. So the production and mock datastore import the same package where `datastore.Entity` and `datastore.Store` are defined. As Enitity and Store are both interfaces there is no real code changes, just shuffling between packages.



> Should this not be `package datastore`?